### PR TITLE
Make line info public and add line number and original line number to it.

### PR DESCRIPTION
**Pull request by SetoKaiba**
Make line info public and add line number and original line number to it.
More info is in the thread.

https://intellij-support.jetbrains.com/hc/en-us/community/posts/360006634979-I-want-to-extend-the-Annotate-view-of-Git-VCS-But-?page=1#community_comment_360000966480

### CODE REVIEW

**Comment by SetoKaiba** [comment]
Force push a modification with latest master. No conflicts now.

**Comment by SetoKaiba** [comment]
>But it may break some plugins if they're use LineInfo with reflection before.
If LineInfo delegates its methods to CommitInfo, these plugins should not be broken.
(Unless they use reflection to access fields instead of calling getters)

Add delegate to CommitInfo in LineInfo.

File: plugins/git4idea/src/git4idea/annotate/GitFileAnnotation.java [hunk 1]
```diff
@@ -77,14 +77,14 @@ public String doGetValue(LineInfo info) {
   private final LineAnnotationAspect REVISION_ASPECT = new GitAnnotationAspect(LineAnnotationAspect.REVISION, false) {
     @Override
     protected String doGetValue(LineInfo lineInfo) {
-      return lineInfo.getRevisionNumber().getShortRev();
+      return lineInfo.getCommitInfo().getRevisionNumber().getShortRev();
```
Comment (AMPivovarov) [inline plugins/git4idea/src/git4idea/annotate/GitFileAnnotation.java]: None of these `getCommitInfo` seems to be necessary.

File: plugins/git4idea/src/git4idea/annotate/GitFileAnnotation.java [hunk 2]
```diff
@@ -265,12 +265,12 @@ protected void showAffectedPaths(int lineNum) {
...
         LineInfo info = myLines.get(lineNum);
 
-        AbstractVcsHelperImpl.loadAndShowCommittedChangesDetails(myProject, info.getRevisionNumber(), myFilePath, () -> getRevisionsChangesProvider().getChangesIn(lineNum));
+        AbstractVcsHelperImpl.loadAndShowCommittedChangesDetails(myProject, info.getCommitInfo().getRevisionNumber(), myFilePath, () -> getRevisionsChangesProvider().getChangesIn(lineNum));
       }
     }
   }
 
-  public static class LineInfo {
+  public static class CommitInfo {
```
Comment (AMPivovarov) [inline plugins/git4idea/src/git4idea/annotate/GitFileAnnotation.java]: Is there are a reason, why `CommitInfo` should be public?
Its methods are duplicated by `LineInfo`.

File: plugins/git4idea/src/git4idea/annotate/GitFileAnnotation.java [hunk 3]
```diff
@@ -343,6 +348,76 @@ public String getSubject() {
...
+    public int getLineNumber() {
+      return myLineNumber;
+    }
+
+    public int getOriginalLineNumber() {
+      return myOriginalLineNumber;
+    }
+
+    @NotNull
+    public Project getProject() {
```
Comment (AMPivovarov) [inline plugins/git4idea/src/git4idea/annotate/GitFileAnnotation.java]: Why one might need `LineInfo.getProject` over `FileAnnotation.getProject` ?

**Comment by SetoKaiba** [comment]
Force push a new commit. It follows the reviews.

**Comment by klikh** [comment]
@SetoKaiba 
Have you submitted a contributor license agreement? If not, please follow the steps described at http://www.jetbrains.com/agreements/cla/ to sign it.

**Comment by SetoKaiba** [comment]
Signed.

**Pull request closed without merging**